### PR TITLE
Option to make x-axis calendar-scaled

### PR DESCRIPTION
**Pull request by ankurjuneja**
#### Rationale
Calendar-based X-axis grouping for QC trend plots: a third option alongside "per replicate" and "per date" that spaces acquisition dates along a real elapsed-time axis, so gaps between runs  are visible instead of being collapsed into evenly-spaced ticks. Same-day replicates still jitter together and the endpoints/extent match per-date mode, so it reads as a familiar variant  rather than a new plot.

https://github.com/LabKey/platform/pull/7883

#### Related Pull Requests
* https://github.com/LabKey/platform/pull/7740

#### Changes
* Added "calendar" X-axis grouping radio option (calendarX state).   
* Grouping change re-renders from cache, no refetch.  

#### Tasks 📍
- [x] Claude Code Review
- [ ] Manual Testing
- [x] Test Automation
- [ ] Verify Fix

### CODE REVIEW

File: webapp/TargetedMS/js/QCTrendPlotPanel.js [hunk 1]
```diff
@@ -1014,28 +1022,76 @@ Ext4.define('LABKEY.targetedms.QCTrendPlotPanel', {
...
         if (!this.groupedXRadioGroup) {
-            this.groupedXRadioGroup = Ext4.create('Ext.form.RadioGroup', {
+            const me = this;
+            const onChange = function(radio, checked) {
+                if (!checked) {
+                    return;
+                }
+                const val = radio.inputValue;
+                const newCalendarX = val === 'calendar';
+                const newGroupedX = val === 'date' || val === 'calendar';
```
Comment (labkey-jeckels) [inline webapp/TargetedMS/js/QCTrendPlotPanel.js]: Constants?

**Comment by labkey-jeckels** [comment]
This is nice. I like how you can clearly see weekends in some of my testing data, for example.

Some issues I found in manual testing:

- [x] This doesn't work with the "always show" option for the guide set. All data is being shown.
- [x] There are scenarios where we cram too many date labels onto the axis. This is probably not unique to this x-axis configuration. We should omit some of the labels when we can't have at least a little separation between the text. See below.
- [x] Our algorithm for dropping labels on the x-axis when it's too crowded is strange when there's a lot of dates and data. See second screenshot below. In this mode, it would be better to show ticks on a consistent basis, like the first of the month or every X days.
- [x] We should draw a continuous line for the SD and mean lines. They are disjoint when there is a gap, and there isa tiny gap between days too. See below.


image

image

**Comment by labkey-jeckels** [comment]
image

image

**Comment by labkey-jeckels** [comment]
This is not working well for me when I have a guide set, choose the "always show" it option and then view the data in a big QC folder. There's no difference between the "per date" and "calendar" renderings. They don't show the jittered points, even when I'm zoomed in to the most recent data.

image

**Comment by ankurjuneja** [comment]
This is how it looks with latest fixes:

https://github.com/user-attachments/assets/878295ff-3c53-45d1-8be6-ff0170404b03

**Comment by ankurjuneja** [comment]
@labkey-jeckels Calendar now differs from per-date but~70% of the plot is an empty gap with a single connecting line. How should calendar mode handle the large no-data gap between the guideset and the date-range window?

**Comment by labkey-jeckels** [comment]
> @labkey-jeckels Calendar now differs from per-date but~70% of the plot is an empty gap with a single connecting line. How should calendar mode handle the large no-data gap between the guideset and the date-range window?

Here's an idea to try in all three x-axis modes when the guide set is outside of the selected date range but displayed, inject a "blank" date entry and x-axis tick/label that corresponds with the first date in the selected range if that date doesn't have a replicate.

That will help explain the big gap in calendar mode, and helps better define the "main" dates plotted vs the guide set dates.

Does that make sense? If not, let's meet next week to discuss/brainstorm.

**Comment by labkey-jeckels** [comment]
Found one problem in my latest testing:

For a folder with lots of data, I see the individual data points when viewing ranges like "Last 7 Days" when I don't have the guide set shown. They disappear when I add the guide set. We should show them in both setups. It's expected that they disappear when I do a long date range, with or without the guide set showing.

image

image

**Comment by labkey-jeckels** [comment]
I am also seeing a JS console error. I haven't figured out the specific repro yet:

Error: <g> attribute transform: Expected number, "translate(undefined,65)".
d3-3.5.17.js?1142232046:670 Error: <g> attribute transform: Expected number, "translate(undefined,65)".
attrFunction @ d3-3.5.17.js?1142232046:670
(anonymous) @ d3-3.5.17.js?1142232046:962
d3_selection_each @ d3-3.5.17.js?1142232046:968
d3_selectionPrototype.each @ d3-3.5.17.js?1142232046:961
d3_selectionPrototype.attr @ d3-3.5.17.js?1142232046:652
addAnnotationsToPlot @ QCTrendPlotPanel.js?1142232046:2702
addEachCombinedPrecursorPlot @ QCPlotHelperBase.js?1142232046:1018
addCombinedPeptideSinglePlot @ QCPlotHelperWrapper.js?1142232046:199
renderPlots @ QCPlotHelperBase.js?1142232046:480
processPlotData @ QCPlotHelperBase.js?1142232046:401
success @ QCPlotHelperBase.js?1142232046:147
callback @ Ajax.ts:130
(anonymous) @ Ajax.ts:322
XMLHttpRequest.send
(anonymous) @ Ajax.ts:346
getPlotsData @ QCPlotHelperBase.js?1142232046:138
processAnnotationData @ QCTrendPlotPanel.js?1142232046:1547
success @ QCTrendPlotPanel.js?1142232046:1500
...

**Comment by labkey-jeckels** [comment]
> I am also seeing a JS console error. I haven't figured out the specific repro yet:
> 
> Error: attribute transform: Expected number, "translate(undefined,65)". d3-3.5.17.js?1142232046:670 Error: attribute transform: Expected number, "translate(undefined,65)". attrFunction @ d3-3.5.17.js?1142232046:670 (anonymous) @ d3-3.5.17.js?1142232046:962 d3_selection_each @ d3-3.5.17.js?1142232046:968 d3_selectionPrototype.each @ d3-3.5.17.js?1142232046:961 d3_selectionPrototype.attr @ d3-3.5.17.js?1142232046:652 addAnnotationsToPlot @ QCTrendPlotPanel.js?1142232046:2702 addEachCombinedPrecursorPlot @ QCPlotHelperBase.js?1142232046:1018 addCombinedPeptideSinglePlot @ QCPlotHelperWrapper.js?1142232046:199 renderPlots @ QCPlotHelperBase.js?1142232046:480 processPlotData @ QCPlotHelperBase.js?1142232046:401 success @ QCPlotHelperBase.js?1142232046:147 callback @ Ajax.ts:130 (anonymous) @ Ajax.ts:322 XMLHttpRequest.send (anonymous) @ Ajax.ts:346 getPlotsData @ QCPlotHelperBase.js?1142232046:138 processAnnotationData @ QCTrendPlotPanel.js?1142232046:1547 success @ QCTrendPlotPanel.js?1142232046:1500 ...

I pushed a change that appears to fix this.
